### PR TITLE
[FLINK-34271][table-planner] fix the potential failure test about GroupAggregateRestoreTest#AGG_WITH_STATE_TTL_HINT

### PR DESCRIPTION
**Pull request by xuyangzhong**
## What is the purpose of the change

This pr tries to fix the Due to state TTL, the state in the metadata savepoint may expire. Therefore, we should not rely on the state data in the savepoint to test.

## Brief change log

  - *Remove the dependency on the state data in the savepoint when testing*

## Verifying this change

The modified test can cover this pr.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented?

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupAggregateTestPrograms.java [hunk 1]
```diff
@@ -383,11 +399,10 @@ public class GroupAggregateTestPrograms {
                                             "avg_a DOUBLE",
                                             "min_c VARCHAR",
                                             "PRIMARY KEY (b) NOT ENFORCED")
-                                    .consumedBeforeRestore(
+                                    .consumedValues(
```
Comment (LadyForest) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupAggregateTestPrograms.java:402]: What's the point here?`consumedValues` invokes `consumedBeforeRestore`.

Comment (xuyangzhong) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupAggregateTestPrograms.java:402]: I think even though they internally use the same method, my intention in using `consumedValues` is to indicate that this test is not concerned with the data before and after the restore.

Comment (LadyForest) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupAggregateTestPrograms.java:402]: My concern is this test spec does not test the `state_ttl` hint. Even if we remove the hint, the result remains unchanged.

Comment (xuyangzhong) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/GroupAggregateTestPrograms.java]: I have updated the state ttl hint to "1s" to let state in savepoint always has expired at first.

**Comment by xuyangzhong** [comment]
@flinkbot run azure

**Comment by xuyangzhong** [comment]
The failure is caused by https://issues.apache.org/jira/browse/FLINK-34272

**Review by LadyForest — APPROVED** [review]
The CI failure is irrelevant to the PR. Will merge

**Pull request merged**